src/com/utils/SudokuGenerator.java

```diff
@@ -0,0 +1,189 @@
+package com.utils;
+
+import com.algorithm.Algorithm;
+import com.algorithm.Backtracking;
+import com.sudoku.Cell;
+import com.sudoku.SudokuBoard;
+
+import java.util.ArrayList;
+import java.util.HashMap;
+import java.util.List;
+import java.util.Map;
+
+/**
+ * Class that represent the Sudoku Generator class
+ */
+public class SudokuGenerator {
+
+    private static SudokuBoard board = new SudokuBoard();
+
+    public enum Level {
+        EASY,
+        MEDIUM,
+        HARD
+    }
+
+    public static Map<Level, Integer> levels = new HashMap<Level, Integer>();
+
+    static {
+        levels.put(Level.EASY, 3);   // max 3 empty cells per sub grid
+        levels.put(Level.MEDIUM, 5); // max 5 empty cells per sub grid
+        levels.put(Level.HARD, 7);   // max 7 empty cells per sub grid
+    }
+
+    public static void generateEasy() {
+        generate(levels.get(Level.EASY));
+    }
+
+    public static void generateMedium() {
+        generate(levels.get(Level.MEDIUM));
+    }
+
+    public static void generateHard() {
+        generate(levels.get(Level.HARD));
+    }
+
+    private static void generate(Level level) {
+        generate(levels.get(level));
+    }
+
+    /**
+     * Generate a board according to a complexity
+     *
+     * @param complexity The max empty cells per subGrid
+     * @return The sudoku board generated
+     */
+    public static void generate(int complexity) {
+        setOneCellOnSubGrids();
+
+        Algorithm backtracking = new Backtracking();
+        backtracking.solve(board);
+
+        clearNCellsOnSubGrids(complexity);
+
+        exportSudokuGame();
+    }
+
+    /**
+     * Given there are nine sub grids on the board
+     * numbered from 0 to 8, this method set one cell on each sub grid
+     * randomly
+     */
+    public static void setOneCellOnSubGrids() {
+        ArrayList<Integer> subGridOrder = getSubGridsOrder();
+        for (Integer subGrid : subGridOrder) {
+            setOneCellOnSubGrid(subGrid);
+        }
+    }
+
+    /**
+     * Given there are nine sub grids on the board
+     * numbered from 0 to 8, this method clear n cells on each sub grid
+     * randomly
+     */
+    public static void clearNCellsOnSubGrids(int complexity) {
+        ArrayList<Integer> subGridOrder = getSubGridsOrder();
+        for (Integer subGrid : subGridOrder) {
+            clearNCellsOnSubGrid(subGrid, getComplexity(complexity));
+        }
+    }
+
+    /**
+     * Given there are nine cells on a sub grid
+     * numbered from 0 to 8, this method gives a group of cell numbers
+     * from sub grid according to the size parameter randomly
+     *
+     * @return The cell number randomly
+     */
+    private static int getComplexity(int maxComplexity) {
+        int minComplexity = maxComplexity - 2;
```

Comment:
What it means that "2" that is decreasing the maxComplexity, looks like a magic number